iroha/src/tx.rs

```diff
@@ -181,7 +181,7 @@ impl AcceptedTransaction {
             .duration_since(SystemTime::UNIX_EPOCH)
             .expect("Failed to get System Time.");
 
-        (current_time - Duration::from_millis(self.payload.creation_time))
+        current_time.saturating_sub(Duration::from_millis(self.payload.creation_time))
```

Comment:
Is it ok though? I think we should check if we have saturation then we can expire it (so discourage peers and clients to do like that)

But that might be a bad idea, as we can drop transactions if our time is not synced

Reply:
I think the same as your last line.
"Overflowed then expire" strategy may refuse a transaction which is enough young.
Can we accept that?
Overflow itself is unhealthy but it does not mean the transaction is invalid

Reply:
Thing is that this is a wicked problem (if we dont have a time sync in a consensus):
- on one hand you can have unsync time and drop valid transactions
- you can be easily flooded with lots of transactions which never expire

Reply:
I'm thinking about what if someone cheat `tx.payload.creation_time` into future time.
And there must have been the log about why TTL is required for transactions, where is it?

Reply:
As it is in `is_expired` function - saturating sub is ok. To prevent spam as you suggest @i1i1 we can add a check in the `from_transaction` function. E.g. drop return Err in `from_transaction` if current_time is less then tx creation time.

Can you add it here @s8sato ?

Reply:
@eadventurous the issue might be that we have unsync time. Correcting time is kinda long process. In this case we might drop valid transactions

Reply:
Ok, in this case it is better to proceed without this check for now.

Reply:
But we will have it in mind for when we have time sync. Also introducing account nonce might be a good idea, but we should discuss it separately.

Reply:
I suggest then to merge this PR as is.